osbenchmark/aggregator.py

```diff
@@ -0,0 +1,216 @@
+from typing import Any, Dict, List, Union
+import uuid
+
+from osbenchmark.metrics import FileTestExecutionStore
+from osbenchmark import metrics, workload, config
+
+class Aggregator:
+    def __init__(self, cfg, test_executions_dict):
+        self.config = cfg
+        self.test_executions = test_executions_dict
+        self.accumulated_results: Dict[str, Dict[str, List[Any]]] = {}
+        self.accumulated_iterations: Dict[str, int] = {}
+
+    # count iterations for each operation in the workload
+    def iterations(self) -> None:
+        loaded_workload = workload.load_workload(self.config)
+        for task in loaded_workload.test_procedures:
```

Comment:
I think `loaded_workload.test_procedures` returns a list of test procedures instead of an instance of test procedure. If that's the case, we should update line 17 to use `test_procedure` instead of task. We should also update line 18 to use `operation` to `task` and from `task.schedule` to `test_procedure.schedule`

Reply:
if the test_procedure matches the test_procedures from the test execution ids, we'll get those tasks / operations

Reply:
> if the test_procedure matches the test_procedures from the test execution ids, we'll get those tasks / operations
Missed this comment, will add a check for this 👍